[TEST-FAILURE] ClientExecutionPoolSizeLowTest.testNodeTerminateWithAsyncOperations

https://hazelcast-l337.ci.cloudbees.com/view/Hazelcast/job/Hazelcast-3.x-nightly/com.hazelcast$hazelcast-client/247/testReport/junit/com.hazelcast.client.io/ClientExecutionPoolSizeLowTest/testNodeTerminateWithAsyncOperations/

```
java.lang.RuntimeException: com.hazelcast.core.HazelcastException: com.hazelcast.spi.exception.RetryableIOException: Packet not send to -> Address[127.0.0.1]:5702
    at com.hazelcast.util.ExceptionUtil.rethrow(ExceptionUtil.java:52)
    at com.hazelcast.util.ExceptionUtil.rethrow(ExceptionUtil.java:47)
    at com.hazelcast.client.spi.ClientProxy.invoke(ClientProxy.java:139)
    at com.hazelcast.client.proxy.ClientMapProxy.size(ClientMapProxy.java:932)
    at com.hazelcast.test.HazelcastTestSupport$3.run(HazelcastTestSupport.java:121)
    at com.hazelcast.test.HazelcastTestSupport.assertTrueEventually(HazelcastTestSupport.java:235)
    at com.hazelcast.test.HazelcastTestSupport.assertSizeEventually(HazelcastTestSupport.java:118)
    at com.hazelcast.test.HazelcastTestSupport.assertSizeEventually(HazelcastTestSupport.java:114)
    at com.hazelcast.client.io.ClientExecutionPoolSizeLowTest.testNodeTerminateWithAsyncOperations(ClientExecutionPoolSizeLowTest.java:84)
Caused by: com.hazelcast.spi.exception.RetryableIOException: Packet not send to -> Address[127.0.0.1]:5702
    at com.hazelcast.spi.impl.BasicInvocation.doInvokeRemote(BasicInvocation.java:297)
    at com.hazelcast.spi.impl.BasicInvocation.doInvoke(BasicInvocation.java:270)
    at com.hazelcast.spi.impl.BasicInvocation.run(BasicInvocation.java:569)
    at com.hazelcast.spi.impl.BasicPartitionInvocation.run(BasicPartitionInvocation.java:28)
    at com.hazelcast.util.executor.CachedExecutorServiceDelegate$Worker.run(CachedExecutorServiceDelegate.java:209)
    at java.util.concurrent.ThreadPoolExecutor.runWorker(ThreadPoolExecutor.java:1145)
    at java.util.concurrent.ThreadPoolExecutor$Worker.run(ThreadPoolExecutor.java:615)
    at java.lang.Thread.run(Thread.java:744)
    at com.hazelcast.util.executor.PoolExecutorThreadFactory$ManagedThread.run(PoolExecutorThreadFactory.java:59)
    at ------ End remote and begin local stack-trace ------.(Unknown Source)
    at com.hazelcast.spi.impl.BasicInvocationFuture.resolveApplicationResponse(BasicInvocationFuture.java:349)
    at com.hazelcast.spi.impl.BasicInvocationFuture.resolveApplicationResponseOrThrowException(BasicInvocationFuture.java:282)
    at com.hazelcast.spi.impl.BasicInvocationFuture.get(BasicInvocationFuture.java:181)
    at com.hazelcast.spi.impl.BasicInvocationFuture.get(BasicInvocationFuture.java:160)
    at com.hazelcast.spi.impl.BasicOperationService$InvokeOnPartitions.retryFailedPartitions(BasicOperationService.java:512)
    at com.hazelcast.spi.impl.BasicOperationService$InvokeOnPartitions.invoke(BasicOperationService.java:450)
    at com.hazelcast.spi.impl.BasicOperationService$InvokeOnPartitions.access$600(BasicOperationService.java:420)
    at com.hazelcast.spi.impl.BasicOperationService.invokeOnAllPartitions(BasicOperationService.java:286)
    at com.hazelcast.client.AllPartitionsClientRequest.process(AllPartitionsClientRequest.java:32)
    at com.hazelcast.client.ClientEngineImpl$ClientPacketProcessor.processRequest(ClientEngineImpl.java:391)
    at com.hazelcast.client.ClientEngineImpl$ClientPacketProcessor.run(ClientEngineImpl.java:334)
    at java.util.concurrent.ThreadPoolExecutor.runWorker(ThreadPoolExecutor.java:1145)
    at java.util.concurrent.ThreadPoolExecutor$Worker.run(ThreadPoolExecutor.java:615)
    at java.lang.Thread.run(Thread.java:744)
    at com.hazelcast.util.executor.PoolExecutorThreadFactory$ManagedThread.run(PoolExecutorThreadFactory.java:59)
```

--- a/hazelcast/src/main/java/com/hazelcast/nio/SocketAcceptor.java
+++ b/hazelcast/src/main/java/com/hazelcast/nio/SocketAcceptor.java
@@ -133,7 +133,7 @@ public class SocketAcceptor implements Runnable {
                 memberSocketInterceptor.onAccept(socketChannel.socket());
             }
             socketChannel.configureBlocking(false);
-            connectionManager.assignSocketChannel(socketChannel);
+            connectionManager.assignSocketChannel(socketChannel, null);
         } catch (Exception e) {
             log(Level.WARNING, e.getClass().getName() + ": " + e.getMessage(), e);
             IOUtil.closeResource(socketChannel);

--- a/hazelcast/src/main/java/com/hazelcast/nio/SocketConnector.java
+++ b/hazelcast/src/main/java/com/hazelcast/nio/SocketConnector.java
@@ -133,7 +133,7 @@ public class SocketConnector implements Runnable {
             }
 
             socketChannelWrapper.configureBlocking(false);
-            TcpIpConnection connection = connectionManager.assignSocketChannel(socketChannelWrapper);
+            TcpIpConnection connection = connectionManager.assignSocketChannel(socketChannelWrapper, address);
             connection.getWriteHandler().setProtocol(Protocols.CLUSTER);
             connectionManager.sendBindRequest(connection, address, true);
         } catch (Exception e) {

--- a/hazelcast/src/main/java/com/hazelcast/nio/TcpIpConnectionManager.java
+++ b/hazelcast/src/main/java/com/hazelcast/nio/TcpIpConnectionManager.java
@@ -307,13 +307,16 @@ public class TcpIpConnectionManager implements ConnectionManager {
         return socketChannelWrapper;
     }
 
-    TcpIpConnection assignSocketChannel(SocketChannelWrapper channel) {
+    TcpIpConnection assignSocketChannel(SocketChannelWrapper channel, Address endpoint) {
         final int index = nextSelectorIndex();
-        final TcpIpConnection connection = new TcpIpConnection(this, inSelectors[index], outSelectors[index], connectionIdGen.incrementAndGet(), channel);
+        final TcpIpConnection connection = new TcpIpConnection(this, inSelectors[index],
+                outSelectors[index], connectionIdGen.incrementAndGet(), channel);
+        connection.setEndPoint(endpoint);
         activeConnections.add(connection);
         acceptedSockets.remove(channel);
         connection.getReadHandler().register();
-        log(Level.INFO, channel.socket().getLocalPort() + " accepted socket connection from " + channel.socket().getRemoteSocketAddress());
+        log(Level.INFO,  "Established socket connection between " + channel.socket().getLocalSocketAddress()
+                + " and " + channel.socket().getRemoteSocketAddress());
         return connection;
     }
 

